sn_client/src/api.rs

```diff
@@ -78,7 +79,8 @@ impl Client {
         #[cfg(feature = "open-metrics")]
         network_builder.metrics_registry(Registry::default());
 
-        let (network, mut network_event_receiver, swarm_driver) = network_builder.build_client()?;
+        let (network, mut network_event_receiver, swarm_driver) =
+            network_builder.build_client(joins_gossip)?;
```

Comment:
Just a small nit, but can we add `enable_gossipsub` into `NetworkBuilder`? And let it be `false` by default and enabled just for nodes?